addons/sale/controllers/portal.py

```diff
@@ -214,7 +214,7 @@ def portal_order_page(self, order_id, report_type=None, access_token=None, messa
                 'fees_by_acquirer': fees_by_acquirer,
                 'show_tokenize_input': show_tokenize_input,
                 'amount': order_sudo.amount_total,
-                'currency': order_sudo.pricelist_id.currency_id,
+                'currency': order_sudo.pricelist_id.currency_id or order_sudo.currency_id,
```

Comment:
```suggestion
                'currency': order_sudo.currency_id,
```
Always use the order currency